src/Gui/Clipping.cpp

```diff
@@ -199,51 +282,103 @@ Clipping* Clipping::makeDockWidget(Gui::View3DInventor* view)
     return clipping;
 }
 
+void Clipping::switchView(Gui::View3DInventor const* view)
+{
+    auto p = d.value(view);
+    if (!p) {
+        p = new Private();
+        p->view = view;
+        setupPrivate(p);
+    }
+    Private::activeClip = p;
+    switchUi(p);
+}
+
 /** Destroys the object and frees any allocated resources */
 Clipping::~Clipping()
 {
-    d->node->removeChild(d->clipX);
-    d->node->removeChild(d->clipY);
-    d->node->removeChild(d->clipZ);
-    d->node->removeChild(d->clipView);
-    d->node->unref();
-    delete d;
+    // clang-format off
+    disconnect(Private::ui.groupBoxX, &QGroupBox::toggled,
```

Comment:
This is a lot of unnecessary code - these disconnects were not needed when it was in Private instance, and won't be needed if we merge Private into Clipping.